cfme/tests/cloud/test_providers.py

```diff
@@ -1215,52 +1215,52 @@ def test_refresh_with_stack_without_parameters(provider, request, stack_without_
     provider.validate_stats(ui=True)
 
 
-@test_requirements.ec2
-@pytest.mark.manual
-def test_ec2_public_images():
+@test_requirements.cloud
+@pytest.mark.long_running
+@pytest.mark.meta(automates=[1491330, 1612086])
+@pytest.mark.provider([AzureProvider, EC2Provider], scope="function")
+def test_public_images_enable_disable(setup_provider, request, appliance, provider):
     """
+    Bugzilla:
+        1491330
+        1612086
+    The easiest way to simulate AWS API Limit for > 200 items is to enable
+    and disable public images.
+    So test for testing public images and for testing AWS API Limit is combined in this test.
     Polarion:
         assignee: mmojzis
         caseimportance: critical
-        initialEstimate: 2/3h
+        initialEstimate: 1 1/2h
         casecomponent: Cloud
         testSteps:
             1. Enable public images for ec2
             2. Add ec2 provider
             3. Wait for its refresh(It can take more than 30 minutes)
+            4. Disable public images for ec2
+            5. Wait for its refresh(It can take more than 30 minutes)
         expectedResults:
             1.
             2.
-            3. Refresh should be successful and images collected
-    """
-    pass
-
-
-@test_requirements.ec2
-@pytest.mark.manual
-def test_ec2_api_filter_limit():
-    """
-    Bugzilla:
-        1612086
-
-    The easiest way to simulate AWS API Limit for > 200 items is to enable
-    and disable public images:
-    Requirement: Have an ec2 provider
-
-    Polarion:
-        assignee: mmojzis
-        casecomponent: Cloud
-        initialEstimate: 1 1/3h
-        startsin: 5.9
-        caseimportance: critical
-        testSteps:
-            1. Enable public images for ec2 in Advanced Settings
-            2. Disable public images for ec2 in Advanced Settings
-        expectedResults:
-            1. Wait for public images to be refreshed
-            2. Wait for public images to be refreshed (cleared)
-    """
-    pass
+            3. Refresh should be successful and public images collected
+            4.
+            5. Refresh should be successful and public images uncollected
+    """
+    # if provider gets stuck loading images it could take more than two hours to be in operating
+    # state which can cause other test to fail so better to delete provider for safety
+    request.addfinalizer(lambda: provider.delete_if_exists())
+    request.addfinalizer(lambda: appliance.set_public_images(provider, enabled=False))
+    # enable
+    public_provider_images_min = 20000 if provider.one_of(AzureProvider) else 40000
+    private_provider_images_max = 5000
+    appliance.set_public_images(provider, enabled=True)
+    provider.refresh_provider_relationships(method='ui')
+    wait_for(lambda: int(provider.load_details(refresh=True).entities.summary("Relationships")
+        .get_text_of("Images")) > public_provider_images_min, delay=120, timeout=3600 * 3)
```

Comment:
This performance is terrible, please write an upstream issue here to record it:
https://github.com/ManageIQ/manageiq-providers-amazon/issues

Otherwise, thank you for automating this, but we don't want to spend so much time on downstream builds testing this when the provider testing is deprecated.

Please add ignore_stream marker so this is only collected against upstream.

Thanks @mmojzis !

Reply:
I already reported this behavior downstream two years ago. There is no way to speed it up.

Reply:
@mmojzis Can you add the `ignore_stream` marker so that this only runs against upstream?

Reply:
Oops, sorry...Added it to different test :man_facepalming: Now it should be okay...Don't know why travis is failing...

Reply:
@mmojzis I think you'll have to do 
```
@pytest.mark.ignore_stream("5.10")
@pytest.mark.ignore_stream("5.11")
```
The `"downstream"` won't actually ignore anything AFAIK.